Kernel module fails to compile on Kernel 6.10.1 on Fedora 40

Running Fedora 40 with kernel Linux 6.10.1-200.fc40.x86_64, when I use the command make -C /lib/modules/`uname -r`/build M=$PWD the following output is presented and the kernel module fails to build:

andrew@galaxy:~/samsung-galaxybook-extras/driver$ make -C /lib/modules/`uname -r`/build M=$PWD
make: Entering directory '/usr/src/kernels/6.10.1-200.fc40.x86_64'
  CC [M]  /home/andrew/samsung-galaxybook-extras/driver/samsung-galaxybook.o
/home/andrew/samsung-galaxybook-extras/driver/samsung-galaxybook.c:805:6: warning: no previous prototype for ‘galaxybook_fan_speed_exit’ [-Wmissing-prototypes]
  805 | void galaxybook_fan_speed_exit(struct samsung_galaxybook *galaxybook)
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/andrew/samsung-galaxybook-extras/driver/samsung-galaxybook.c:1649:10: error: ‘struct acpi_driver’ has no member named ‘owner’
 1649 |         .owner = THIS_MODULE,
      |          ^~~~~
In file included from ./include/linux/printk.h:6,
                 from ./include/linux/kernel.h:31,
                 from /home/andrew/samsung-galaxybook-extras/driver/samsung-galaxybook.c:13:
./include/linux/init.h:180:21: error: initialization of ‘const char *’ from incompatible pointer type ‘struct module *’ [-Wincompatible-pointer-types]
  180 | #define THIS_MODULE (&__this_module)
      |                     ^
/home/andrew/samsung-galaxybook-extras/driver/samsung-galaxybook.c:1649:18: note: in expansion of macro ‘THIS_MODULE’
 1649 |         .owner = THIS_MODULE,
      |                  ^~~~~~~~~~~
./include/linux/init.h:180:21: note: (near initialization for ‘galaxybook_acpi_driver.drv.name’)
  180 | #define THIS_MODULE (&__this_module)
      |                     ^
/home/andrew/samsung-galaxybook-extras/driver/samsung-galaxybook.c:1649:18: note: in expansion of macro ‘THIS_MODULE’
 1649 |         .owner = THIS_MODULE,
      |                  ^~~~~~~~~~~
/home/andrew/samsung-galaxybook-extras/driver/samsung-galaxybook.c:1639:52: warning: missing braces around initializer [-Wmissing-braces]
 1639 | static struct acpi_driver galaxybook_acpi_driver = {
      |                                                    ^
make[2]: *** [scripts/Makefile.build:244: /home/andrew/samsung-galaxybook-extras/driver/samsung-galaxybook.o] Error 1
make[1]: *** [/usr/src/kernels/6.10.1-200.fc40.x86_64/Makefile:1946: /home/andrew/samsung-galaxybook-extras/driver] Error 2
make: *** [Makefile:252: __sub-make] Error 2
make: Leaving directory '/usr/src/kernels/6.10.1-200.fc40.x86_64'

Kernel 6.9 series is deprecated. Same issue with Kernel 6.10.3-200.fc40.x86_64:

andrew@galaxy:~/samsung-galaxybook-extras/driver$ make -C /lib/modules/`uname -r`/build M=$PWD
make: Entering directory '/usr/src/kernels/6.10.3-200.fc40.x86_64'
  CC [M]  /home/andrew/samsung-galaxybook-extras/driver/samsung-galaxybook.o
/home/andrew/samsung-galaxybook-extras/driver/samsung-galaxybook.c:805:6: warning: no previous prototype for ‘galaxybook_fan_speed_exit’ [-Wmissing-prototypes]
  805 | void galaxybook_fan_speed_exit(struct samsung_galaxybook *galaxybook)
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/andrew/samsung-galaxybook-extras/driver/samsung-galaxybook.c:1649:10: error: ‘struct acpi_driver’ has no member named ‘owner’
 1649 |         .owner = THIS_MODULE,
      |          ^~~~~
In file included from ./include/linux/printk.h:6,
                 from ./include/linux/kernel.h:31,
                 from /home/andrew/samsung-galaxybook-extras/driver/samsung-galaxybook.c:13:
./include/linux/init.h:180:21: error: initialization of ‘const char *’ from incompatible pointer type ‘struct module *’ [-Wincompatible-pointer-types]
  180 | #define THIS_MODULE (&__this_module)
      |                     ^
/home/andrew/samsung-galaxybook-extras/driver/samsung-galaxybook.c:1649:18: note: in expansion of macro ‘THIS_MODULE’
 1649 |         .owner = THIS_MODULE,
      |                  ^~~~~~~~~~~
./include/linux/init.h:180:21: note: (near initialization for ‘galaxybook_acpi_driver.drv.name’)
  180 | #define THIS_MODULE (&__this_module)
      |                     ^
/home/andrew/samsung-galaxybook-extras/driver/samsung-galaxybook.c:1649:18: note: in expansion of macro ‘THIS_MODULE’
 1649 |         .owner = THIS_MODULE,
      |                  ^~~~~~~~~~~
/home/andrew/samsung-galaxybook-extras/driver/samsung-galaxybook.c:1639:52: warning: missing braces around initializer [-Wmissing-braces]
 1639 | static struct acpi_driver galaxybook_acpi_driver = {
      |                                                    ^
make[2]: *** [scripts/Makefile.build:244: /home/andrew/samsung-galaxybook-extras/driver/samsung-galaxybook.o] Error 1
make[1]: *** [/usr/src/kernels/6.10.3-200.fc40.x86_64/Makefile:1946: /home/andrew/samsung-galaxybook-extras/driver] Error 2
make: *** [Makefile:252: __sub-make] Error 2
make: Leaving directory '/usr/src/kernels/6.10.3-200.fc40.x86_64'

Same issue here:

```
make -C /lib/modules/`uname -r`/build M=$PWD
make: Entering directory '/usr/src/kernels/6.10.3-200.fc40.x86_64'
  CC [M]  /home/mhagnumdw/projetos/samsung-galaxybook-extras/driver/samsung-galaxybook.o
/home/mhagnumdw/projetos/samsung-galaxybook-extras/driver/samsung-galaxybook.c:805:6: warning: no previous prototype for ‘galaxybook_fan_speed_exit’ [-Wmissing-prototypes]
  805 | void galaxybook_fan_speed_exit(struct samsung_galaxybook *galaxybook)
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/mhagnumdw/projetos/samsung-galaxybook-extras/driver/samsung-galaxybook.c:1649:10: error: ‘struct acpi_driver’ has no member named ‘owner’
 1649 |         .owner = THIS_MODULE,
      |          ^~~~~
In file included from ./include/linux/printk.h:6,
                 from ./include/linux/kernel.h:31,
                 from /home/mhagnumdw/projetos/samsung-galaxybook-extras/driver/samsung-galaxybook.c:13:
./include/linux/init.h:180:21: error: initialization of ‘const char *’ from incompatible pointer type ‘struct module *’ [-Wincompatible-pointer-types]
  180 | #define THIS_MODULE (&__this_module)
      |                     ^
/home/mhagnumdw/projetos/samsung-galaxybook-extras/driver/samsung-galaxybook.c:1649:18: note: in expansion of macro ‘THIS_MODULE’
 1649 |         .owner = THIS_MODULE,
      |                  ^~~~~~~~~~~
./include/linux/init.h:180:21: note: (near initialization for ‘galaxybook_acpi_driver.drv.name’)
  180 | #define THIS_MODULE (&__this_module)
      |                     ^
/home/mhagnumdw/projetos/samsung-galaxybook-extras/driver/samsung-galaxybook.c:1649:18: note: in expansion of macro ‘THIS_MODULE’
 1649 |         .owner = THIS_MODULE,
      |                  ^~~~~~~~~~~
/home/mhagnumdw/projetos/samsung-galaxybook-extras/driver/samsung-galaxybook.c:1639:52: warning: missing braces around initializer [-Wmissing-braces]
 1639 | static struct acpi_driver galaxybook_acpi_driver = {
      |                                                    ^
make[2]: *** [scripts/Makefile.build:244: /home/mhagnumdw/projetos/samsung-galaxybook-extras/driver/samsung-galaxybook.o] Error 1
make[1]: *** [/usr/src/kernels/6.10.3-200.fc40.x86_64/Makefile:1946: /home/mhagnumdw/projetos/samsung-galaxybook-extras/driver] Error 2
make: *** [Makefile:252: __sub-make] Error 2
make: Leaving directory '/usr/src/kernels/6.10.3-200.fc40.x86_64'
```

--- a/driver/samsung-galaxybook.c
+++ b/driver/samsung-galaxybook.c
@@ -24,6 +24,7 @@
 #include <linux/workqueue.h>
 #include <linux/input.h>
 #include <linux/input/sparse-keymap.h>
+#include <linux/version.h>
 
 #define SAMSUNG_GALAXYBOOK_CLASS  "samsung-galaxybook"
 #define SAMSUNG_GALAXYBOOK_NAME   "Samsung Galaxybook Extras"
@@ -1073,7 +1074,7 @@ static void galaxybook_performance_mode_hotkey_work(struct work_struct *work)
 {
 	struct samsung_galaxybook *galaxybook = container_of(work,
 			struct samsung_galaxybook, performance_mode_hotkey_work);
-	
+
 	platform_profile_set(&galaxybook->profile_handler,
 			performance_mode_hotkey_next_profile[galaxybook->current_profile]);
 
@@ -1646,7 +1647,10 @@ static struct acpi_driver galaxybook_acpi_driver = {
 		.remove = galaxybook_acpi_remove,
 		.notify = galaxybook_acpi_notify,
 		},
+
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 10, 0)
 	.owner = THIS_MODULE,
+#endif
 };
 
 static int __init samsung_galaxybook_init(void)

